proposals/2023/03-15_worker-auth.md

```diff
@@ -0,0 +1,177 @@
+
+This document is a proposal to utilize the recently introduced `internal/token` package to mint and validate worker auth tokens and include a registration process to introduce significant friction for potential attackers.
+
+In short, the Vela platform administrators will have the ability to register workers with a registration token, and the worker will maintain its authenticity by performing a check-in at a specified interval, where it will present its authentication token and receive a new one. The details of this will all be included below, in the [Design](03-12_worker-auth.md#design) section.
+
+This feature is intended to be optional. If platform admins would rather continue to use the symmetric token for the check in process as well as getting build tokens, that will be available.
+
+
+**Please briefly answer the following questions:**
+
+1. Why is this required?
+
+<!-- Answer here -->
+
+In the current implementation, the server-worker symmetric token has no expiration and is not encrypted in any way. 
+
+If a bad actor were to gain access to the host on which the Vela worker is run, they could read from the docker environment and use the symmetric secret to request build tokens, update worker information, and spin up new workers of their own.
+
+2. If this is a redesign or refactor, what issues exist in the current implementation?
+
+<!-- Answer here -->
+
+Mentioned in question 1
+
+3. Are there any other workarounds, and if so, what are the drawbacks?
+
+<!-- Answer here -->
+
+While there may be other forms of authentication available for the server and worker, there already exists the infrastructure to use JWT tokens with claims on expiration, token type, etc.
+
+4. Are there any related issues? Please provide them below if any exist.
+
+<!-- Answer here -->
+
+N/A
+
+## Design
+
+<!--
+This section is intended to explain the solution design for the proposal.
+
+NOTE: If there are no current plans for a solution, please leave this section blank.
+-->
+
+**Please describe your solution to the proposal. This includes, but is not limited to:**
+
+* new/updated endpoints or url paths
+* new/updated configuration variables (environment, flags, files, etc.)
+* performance and user experience trade-offs
+* security concerns or assumptions
+* examples or (pseudo) code snippets
+
+<!-- Answer here -->
+
+A diagram will be added to this proposal soon to provide a visual aid for the design.
+
+### Server Design
+
+With the implementation of the token manager and build tokens, the ground work is already present to mint and validate worker auth tokens.
+
+Some of the major changes will be:
+
+- Add the ability for the `internal/token` package to mint and validate worker auth / register tokens
+- Add a `GET` registration endpoint, which will be platform admin exclusive. It will return a worker registration token.
```

Comment:
This returns the token the admin will use to give the worker to use as its first token, right?

Each request will return a new token, right? If so, I think this should be a POST instead, as that better matches the HTTP verb's general usage. GETs should generally be cache-able by intermediate hosts.

Reply:
Sounds good, I agree